lib/E2E/LowerRankedShapes.cpp

```diff
@@ -177,12 +229,14 @@ class LowerRankedShapes : public LowerRankedShapesBase<LowerRankedShapes> {
     patterns.insert<LowerConstShapeOp>(context);
     patterns.insert<LowerShapeBroadcastOp>(context);
     patterns.insert<LowerShapeGetExtentOp>(context);
+    patterns.insert<LowerShapeToExtentTensorOp>(context);
```

Comment:
(this review comment was meant to be on the class comment for this pass, but github doesn't let me do this)

The class comment is now out of date. It should now be phrased in terms of extent tensors. Also, if you manually add in a fold here that does `extract_element(tensor_from_elements(x0, x1, ...), n) -> xn` then you will restore the original invariant of the pass I think (all shapes are reduced to individual extent SSA values).

Reply:
Updated. Added a todo about the folder and the original invariant. I'm submitting without that since it now works and I fiddled with something similar for a while last night. Will follow-up post-integrate.